h_program-lang/ast_generic.ml

```diff
@@ -373,27 +389,29 @@ and stmt =
   | DirectiveStmt of directive
 
   (* newscope: in C++/Java/Go *)
-  | Block of stmt list
-  (* EmptyStmt = Block [] *)
+  | Block of stmt list (* TODO: bracket *)
+  (* EmptyStmt = Block [], or separate so can not be matched by $S? *)
 
-  | If of expr * stmt * stmt
-  | While of expr * stmt
-  | DoWhile of stmt * expr
+  | If of tok (* 'if' or 'elif' *) * expr * stmt * stmt
+  | While of tok * expr * stmt
+  | DoWhile of tok * stmt * expr
   (* newscope: *)
-  | For of for_header * stmt
+  | For of tok * for_header * stmt
 
   (* less: could be merged with ExprStmt (MatchPattern ...) *)
   | Switch of tok (* switch or also Select in Go *) * expr * case_and_body list
 
-  | Return of expr option
-  | Continue of expr option | Break of expr option (* todo? switch to label? *)
+  | Return   of tok * expr option
+  (* less: switch to label? but PHP accept integers no? *)
+  | Continue of tok * expr option 
+  | Break    of tok * expr option 
```

Comment:
I do believe this is the important line!  DId you want to add specific test for this or do the existing test cover this ?